pkg/framework/plugins/podlifetime/pod_lifetime.go

```diff
@@ -124,8 +124,16 @@ func (d *PodLifeTime) Deschedule(ctx context.Context, nodes []*v1.Node) *framewo
 	podutil.SortPodsBasedOnAge(podsToEvict)
 
 	for _, pod := range podsToEvict {
-		if !d.handle.Evictor().NodeLimitExceeded(nodeMap[pod.Spec.NodeName]) {
-			d.handle.Evictor().Evict(ctx, pod, evictions.EvictOptions{})
+		shouldEvict, err := d.handle.Evictor().PreEvictionFilter(pod)
```

Comment:
With `PreEvictionFilter` returning only bool, `PreEvictionFilter` can be combined with `podFilter` since there's no difference between invoking `PreEvictionFilter` right after `Filter` or right before `Evict` for this strategy. The same holds for other strategies like failedpods, toomanyrestarts, etc.